packages/headless/src/controllers/commerce/context/cart/headless-cart.ts

```diff
@@ -172,7 +180,25 @@ export function buildCart(engine: CommerceEngine, props: CartProps = {}): Cart {
 
     updateItem(item: CartItem) {
       if (isNewQuantityDifferent(item.productId, item.quantity)) {
-        // TODO LENS-1497: log ec.cartAction; if new quantity > previous, 'add', otherwise, 'remove'.
+        const prevItem = itemSelector(getState(), item.productId);
+
+        const {quantity: currentQuantity, ...product} = item;
+        const action = isCurrentItemQuantityGreater(item, prevItem)
+          ? 'add'
+          : 'remove';
+        const quantity = !prevItem
+          ? currentQuantity
+          : Math.abs(currentQuantity - prevItem.quantity);
+        const currency = engine.state.commerceContext
+          .currency as CurrencyCodeISO4217;
+
+        const payload: Ec.CartAction = {
+          action,
+          currency,
+          quantity,
+          product,
+        };
+        engine.relay.emit('ec.cartAction', payload);
       }
 
       dispatch(updateItem(item));
```

Comment:
A few todo:
 - [ ] Move the `if` as a precondition: Essentially, this function doesn't need to run if nothing changed, so let's make it so:
   ```ts
   updateItem(item: CartItem) {
     if (!isNewQuantityDifferent(item.productId, item.quantity)) {
       return;
     }
     //...
   }
   ```
 - [ ] Refactor the payload computation into a separate function `(state, action) => payload`

Reply:
For the first todo:

I don't think it's fully true. If you update the price of the item, you will will have to update the item in the cart (so this condition would prevent this action). However, I think it's a fair point that we should check if there any changes of the between the passed item and the item in the state. I would propose to add a new condition.

Reply:
Maybe it'd be best to have different functions to update item details, such as price, and quantity.
🤔

What do you think @fbeaudoincoveo?
Anyhow, @sallainCoveo , fair enough for now 👌

Reply:
For the second todo:

I did not check it, because I'm not sure if I fully understood the expectation. Should this function be common (we specified the return type, and we construct the payload based on the state/action passed? Or, like I did, I pass as argument the information necessary to construction a specific type of payload (`Ec.CartAction` in that case).